block/manager.go

```diff
@@ -330,12 +396,73 @@ func (m *Manager) trySyncNextBlock(ctx context.Context, daHeight uint64) error {
 		if err != nil {
 			m.logger.Error("failed to save updated state", "error", err)
 		}
-		delete(m.syncCache, currentHeight+1)
+		m.blockCache.deleteBlock(currentHeight + 1)
 	}
 
 	return nil
 }
 
+// BlockStoreRetrieveLoop is responsible for retrieving blocks from the Block Store.
+func (m *Manager) BlockStoreRetrieveLoop(ctx context.Context) {
+	// waitCh is used to signal the block store retrieve loop, that it should check block store for new blocks
+	// blockStoreCond can be signalled in completely async manner, and goroutine below
+	// works as some kind of "buffer" for those signals
+	waitCh := make(chan interface{})
+	lastBlockStoreHeight := uint64(0)
+	go func() {
+		for {
+			// This infinite loop is expected to be stopped once the context is
+			// cancelled or throws an error and cleaned up by the GC. This is OK
+			// because it waits using a conditional which is only signaled periodically.
+			m.blockStoreMtx.Lock()
+			m.blockStoreCond.Wait()
+			waitCh <- nil
+			m.blockStoreMtx.Unlock()
+			if ctx.Err() != nil {
+				return
+			}
+		}
+	}()
+	for {
+		select {
```

Comment:
[non-blocking]
nit: I think it is slightly cleaner to not have the executed code in the select statement

```
select {
  case return:
     return
  case continue:
}

executed code
```

Reply:
That does look cleaner, but I tried doing this and it seems to not be syntactically correct since `return` is a reserved keyword, `case return` is not valid syntax. Is this what this suggestion meant?